packages/@aws-cdk/aws-lambda/lib/image-function.ts

```diff
@@ -41,8 +43,12 @@ export abstract class DockerImageCode {
    */
   public static fromImageAsset(directory: string, props: AssetImageCodeProps = {}): DockerImageCode {
     return {
-      _bind() {
-        return new AssetImageCode(directory, props);
+      _bind(architecture?: Architecture) {
+        return new AssetImageCode(directory, {
+           // determine the platform from `architecture`.
+          ...architecture?.dockerPlatform ? { platform: Platform.custom(architecture.dockerPlatform) } : {},
+          ...props,
+        });
```

Comment:
Still need the unit tests requested here.

Reply:
Sure, I will do it later today